### PR TITLE
Core Extensions for debug

### PR DESCRIPTION
**Pull request by rvais**
I am not sure If this state is exactly the one we left these after our last meeting but it should be.

### CODE REVIEW

File: xmvn-connector-aether/pom.xml [hunk 1]
```diff
@@ -86,6 +91,11 @@
       <artifactId>plexus-container-default</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+    	<groupId>org.apache.maven</groupId>
+    	<artifactId>maven-aether-provider</artifactId>
+    	<version>3.3.9</version>
```
Comment (mizdebsk) [inline xmvn-connector-aether/pom.xml]: Versions shouldn't be hardcoded in POM directly. Use dependencyManagement.

File: xmvn-connector-aether/pom.xml [hunk 2]
```diff
@@ -86,6 +91,11 @@
       <artifactId>plexus-container-default</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
```
Comment (mizdebsk) [inline xmvn-connector-aether/pom.xml]: Scope should be "provided".

File: xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java [hunk 3]
```diff
@@ -291,4 +307,68 @@ public void resolutionCompleted( ResolutionRequest request, ResolutionResult res
...
+        for ( CoreExtension ext : extList )
+        {
+            String[] artifact = { ext.getGroupId(), ext.getArtifactId(), ext.getVersion() };
+            artifacts.add( artifact );
+        }
+    }
+
+    private void logWarningSafely( String msg )
+    {
+        if ( this.logger == null )
```
Comment (mizdebsk) [inline xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java]: logger should never be null. Both "logWarningSafely" functions should be removed.

File: xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java [hunk 4]
```diff
@@ -280,6 +285,17 @@ public void execute()
             }
         }
 
+        if ( extensionSet != null )
```
Comment (mizdebsk) [inline xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java]: extensionSet can't be null at this point.

File: xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java [hunk 5]
```diff
@@ -280,6 +285,17 @@ public void execute()
...
+        {
+            for ( String[] artifactParts : extensionSet )
+            {
+                // String[0] groupId; String[1] artifactId; String[2] version
+
+                Artifact artifact = new DefaultArtifact( artifactParts[0], artifactParts[1], artifactParts[2] );
+                deps.add( new NamespacedArtifact( "", artifact ) );
```
Comment (mizdebsk) [inline xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java]: 1. This logic is not correct. Not all declared extensions should be RPM dependencies, for example some of them might have been resolved from remote repositiories.
2. Don't hardcode namespace to empty string, use actual namespace.
3. You need to set proper artifact version. Version declared in extension list is *not* the same as version of resolved artifact - they can (and often will be) different.

File: xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java [hunk 6]
```diff
@@ -291,4 +307,68 @@ public void resolutionCompleted( ResolutionRequest request, ResolutionResult res
...
+        if ( isXmvnClass )
+        {
+            try
+            {
+                XMvnCoreExtensionsManager xmvnExtManager =
+                    XMvnCoreExtensionsManager.class.cast( this.extensionsManager );
+
+                return xmvnExtManager.getCapturedExtensions();
+            }
+            catch ( ClassCastException e )
```
Comment (mizdebsk) [inline xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java]: Can this actually happen?

File: xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java [hunk 7]
```diff
@@ -291,4 +307,68 @@ public void resolutionCompleted( ResolutionRequest request, ResolutionResult res
...
+            {
+                logWarningSafely( "Unable to gain list of used Maven's core extensions."
+                    + "Operation failed. Extensions will be missing form depency list.", e );
+            }
+        }
+        else
```
Comment (mizdebsk) [inline xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java]: Likewise.

File: xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java (hunk 5 above)
Comment (rvais) [inline xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java]: 1. How should I filter these extensions than ?
2. What name space should I use ? The only reason why I used empty string was, that I dont know what exactli supposed to be their namespace amd I can add name space later.
3. Similarly to 1st point, it requires to resolve extension list against something. Should I use similar code to what you use for normal artifacts ?

**Comment by rvais** [comment]
I have done few of requested changes, for the rest I need more information. Anyway I have refreshed my memory and figured out what was the proplem ve encountered last time (and why I didn't encountered possible issues you pointed up here). The problem is with the injection it self. BootstrapCoreExtensionManager proxy (my XMvnCoreExtensionsManager) is not injected properly. Instead Maven uses it's own original class and that's why you said you need to debug it your self. We couldn't figure out why...

File: xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java (hunk 5 above)
Comment (mizdebsk) [inline xmvn-mojo/src/main/java/org/fedoraproject/xmvn/mojo/BuilddepMojo.java]: > How should I filter these extensions than ?

The same way as dependencies are handled. Iterate over resolved artifacts (`BuilddepMojo.resolutions`) and compare with list of extensions. Resolution which is also extension should be added as build-dependency.

> What name space should I use ? The only reason why I used empty string was, that I dont know what exactli supposed to be their namespace amd I can add name space later.
> Similarly to 1st point, it requires to resolve extension list against something. Should I use similar code to what you use for normal artifacts ?

Compat version and namespace are part of resolution:

    String compatVersion = resolution[1];
    String namespace = resolution[2];

**Comment by mizdebsk** [comment]
I'm closing this PR as it has merge conflicts and there hasn't been any activity in almost 2 years. You are welcome to reopen it when conflicts and other issues are resolved.

**Pull request closed without merging**